### ISSUE TITLE
Retail / Newest TipTac Version-Latest WoW Patch  / 5x TipTac/ttStyle.lua:289: attempt to concatenate a nil value

### ISSUE THREAD

**Issue by Vilogity** [frozn/TipTac#155]
Get this ERROR in Group M+

5x TipTac/ttStyle.lua:289: attempt to concatenate a nil value
[string "@TipTac/ttStyle.lua"]:289: in function `ModifyUnitTooltip'
[string "@TipTac/ttStyle.lua"]:355: in function `?'
[string "@TipTac/ttCore.lua"]:397: in function `SendElementEvent'
[string "@TipTac/ttCore.lua"]:1273: in function `ApplyUnitAppearance'
[string "@TipTac/ttCore.lua"]:1516: in function `OnTooltipSetUnit'
[string "@TipTac/ttCore.lua"]:1870: in function <TipTac/ttCore.lua:1868>
[string "=(tail call)"]: ?
[string "=[C]"]: in function `securecallfunction'
[string "@SharedXML/Tooltip/TooltipDataHandler.lua"]:60: in function <SharedXML/Tooltip/TooltipDataHandler.lua:55>
[string "@SharedXML/Tooltip/TooltipDataHandler.lua"]:80: in function <SharedXML/Tooltip/TooltipDataHandler.lua:76>
[string "@SharedXML/Tooltip/TooltipDataHandler.lua"]:116: in function <SharedXML/Tooltip/TooltipDataHandler.lua:106>
[string "=[C]"]: in function `SetAttribute'
[string "@SharedXML/Tooltip/TooltipDataHandler.lua"]:141: in function <SharedXML/Tooltip/TooltipDataHandler.lua:134>
[string "@SharedXML/Tooltip/TooltipDataHandler.lua"]:282: in function <SharedXML/Tooltip/TooltipDataHandler.lua:240>
[string "=[C]"]: in function `securecallfunction'
[string "@SharedXML/Tooltip/TooltipDataHandler.lua"]:237: in function <SharedXML/Tooltip/TooltipDataHandler.lua:236>
[string "=(tail call)"]: ?
[string "=[C]"]: in function `SetUnit'
[string "@PitBull4/UnitFrame.lua"]:256: in function <PitBull4/UnitFrame.lua:251>

Locals:
self = <table> {
 GenerateTargetLines = <function> defined @TipTac/ttStyle.lua:99
 OnLoad = <function> defined @TipTac/ttStyle.lua:324
 GenerateNpcLines = <function> defined @TipTac/ttStyle.lua:210
 GeneratePetLines = <function> defined @TipTac/ttStyle.lua:179
 GeneratePlayerLines = <function> defined @TipTac/ttStyle.lua:120
 name = "Style"
 ModifyUnitTooltip = <function> defined @TipTac/ttStyle.lua:234
 OnCleared = <function> defined @TipTac/ttStyle.lua:358
 OnStyleTip = <function> defined @TipTac/ttStyle.lua:328
}
u = <table> {
 isPlayer = true
 reactionIndex = 6
 isPetWild = false
 class = "Paladin"
 isPetCompanion = false
 powerType = 0
 token = "party3"
 reactionColor = "|cff25c1eb"
 classFile = "PALADIN"
 name = "Dairybest"
 realm = "TwistingNether"
}
first = true
unit = "party3"
level = 70
classification = "normal"
ratingSummary = <table> {
 runs = <table> {
 }
 currentSeasonScore = 85
}
mythicPlusDungeonScore = 85
mythicPlusBestRunLevel = nil
(*temporary) = <table> {
 1 = "|cffc0c0c0"
 2 = "70"
 3 = " "
 4 = "|cffffffff"
 5 = "Tauren"
 6 = " "
 7 = "|cfff48cba"
 8 = "Paladin"
 9 = "
|cffffd100"
 count = 9
 Index = 2
}
(*temporary) = <function> defined =[C]:-1
(*temporary) = "Wertung für Mythisch+: %s"
(*temporary) = "|cffffffff85|r"
(*temporary) = nil
(*temporary) = "ffffffff"
(*temporary) = "|cffffffff85|r"
(*temporary) = "attempt to concatenate a nil value"
UnitName = <function> defined =[C]:-1
cfg = <table> {
 gttScale = 1
 classification_rare = "%s|r (Rare)"
 classification_elite = "+%s"
 optionsBottom = 567.571289
 tipBackdropEdge = "Interface\AddOns\Quartz\textures\Tooltip-BigBorder"
 optionsLeft = 746.381042
 backdropEdgeSize = 12
 t_showTalents = false
 classification_worldboss = "%s|r (Boss)"
 hideTipsInCombat = "all"
 tipBackdropBG = "Interface\Tooltips\UI-Tooltip-Background"
 classification_trivial = "~%s"
 t_enable = false
 if_iconSize = 40
 targetYouText = "<<YOU>>"
 classification_rareelite = "+%s|r (Rare)"
 pixelPerfectBackdrop = false
 hideUFTipsInCombat = false
 classification_normal = "%s"
 if_spellColoredBorder = true
 showGuildRank = false
 hideAllTipsInCombat = true
 left = 1683.191162
 top = 293.142242
 classification_minus = "-%s"
 classColoredBorder = true
 if_showSpellIdAndRank = false
 if_azeriteEssenceQualityBorder = false
}
UnitIsWildBattlePet = <function> defined =[C]:-1
UnitIsBattlePetCompanion = <function> defined =[C]:-1
lineInfo = <table> {
 1 = "|cffc0c0c0"
 2 = "70"
 3 = " "
 4 = "|cffffffff"
 5 = "Tauren"
 6 = " "
 7 = "|cfff48cba"
 8 = "Paladin"
 9 = "
|cffffd100"
 count = 9
 Index = 2
}
tt = TipTac {
 BottomLeftCorner = Texture {
 }
 close = Button {
 }
 AddModifiedTip = <function> defined @TipTac/ttCore.lua:2153
 MirrorAnchorsSmart = <table> {
 }
 CURSOR_UPDATE = <function> defined @TipTac/ttCore.lua:435
 GetNearestPixelSize = <function> defined @TipTac/ttCore.lua:620
 GetBackdropColor = <function> defined @SharedXML/Backdrop.lua:390
 RightEdge = Texture {
 }
 CVAR_UPDATE = <function> defined @TipTac/ttCore.lua:547
 ClearBackdrop = <function> defined @SharedXML/Backdrop.lua:282
 SetBackdropColorLocked = <function> defined @TipTac/ttCore.lua:796
 ADDON_LOADED = <function> defined @TipTac/ttCore.lua:2040
 PixelSnapDisabled = true
 GetDisplayedUnit = <function> defined @TipTac/ttCore.lua:364
 SetBackdropBorderColorLocked = <function> defined @TipTac/ttCore.lua:811
 ResetBackdropBorderColorLocked = <function> defined @TipTac/ttCore.lua:826
 tipsToModify = <table> {
 }
 SetupTextureCoordinates = <function> defined @SharedXML/Backdrop.lua:214
 SendElementEvent = <function> defined @TipTac/ttCore.lua:394
 HasBackdropInfo = <function> defined @SharedXML/Backdrop.lua:278
 GetEdgeSize = <function> defined @SharedXML/Backdrop.lua:188

**Comment by Vilogity** [comment]
Maybe a Problem with Raider IO Addon showing the Rating ?

**Comment by Dubwize** [comment]
Also seeing this. 
```
21x TipTac/ttStyle.lua:289: attempt to concatenate a nil value
[string "@TipTac/ttStyle.lua"]:289: in function `ModifyUnitTooltip'
[string "@TipTac/ttStyle.lua"]:355: in function `?'
[string "@TipTac/ttCore.lua"]:397: in function `SendElementEvent'
[string "@TipTac/ttCore.lua"]:1273: in function `ApplyUnitAppearance'
[string "@TipTac/ttCore.lua"]:1516: in function `OnTooltipSetUnit'
[string "@TipTac/ttCore.lua"]:1870: in function <TipTac/ttCore.lua:1868>
[string "=(tail call)"]: ?
[string "=[C]"]: in function `securecallfunction'
[string "@SharedXML/Tooltip/TooltipDataHandler.lua"]:60: in function <SharedXML/Tooltip/TooltipDataHandler.lua:55>
[string "@SharedXML/Tooltip/TooltipDataHandler.lua"]:80: in function <SharedXML/Tooltip/TooltipDataHandler.lua:76>
[string "@SharedXML/Tooltip/TooltipDataHandler.lua"]:116: in function <SharedXML/Tooltip/TooltipDataHandler.lua:106>
[string "=[C]"]: in function `SetAttribute'
[string "@SharedXML/Tooltip/TooltipDataHandler.lua"]:141: in function <SharedXML/Tooltip/TooltipDataHandler.lua:134>
[string "@SharedXML/Tooltip/TooltipDataHandler.lua"]:282: in function <SharedXML/Tooltip/TooltipDataHandler.lua:240>
[string "=[C]"]: in function `securecallfunction'
[string "@SharedXML/Tooltip/TooltipDataHandler.lua"]:237: in function `ProcessInfo'
[string "@FrameXML/GameTooltip.lua"]:970: in function `SetWorldCursor'
[string "@FrameXML/UIParent.lua"]:1439: in function <FrameXML/UIParent.lua:1362>

Locals:
self = <table> {
 GenerateTargetLines = <function> defined @TipTac/ttStyle.lua:99
 OnLoad = <function> defined @TipTac/ttStyle.lua:324
 GenerateNpcLines = <function> defined @TipTac/ttStyle.lua:210
 GeneratePetLines = <function> defined @TipTac/ttStyle.lua:179
 GeneratePlayerLines = <function> defined @TipTac/ttStyle.lua:120
 name = "Style"
 ModifyUnitTooltip = <function> defined @TipTac/ttStyle.lua:234
 OnCleared = <function> defined @TipTac/ttStyle.lua:358
 OnStyleTip = <function> defined @TipTac/ttStyle.lua:328
}
u = <table> {
 isPlayer = true
 reactionIndex = 6
 isPetWild = false
 class = "Monk"
 isPetCompanion = false
 powerType = 3
 token = "mouseover"
 reactionColor = "|cff25c1eb"
 classFile = "MONK"
 name = "Spasskid"
 realm = "Tichondrius"
}
first = true
unit = "mouseover"
level = 70
classification = "normal"
ratingSummary = <table> {
 runs = <table> {
 }
 currentSeasonScore = 52
}
mythicPlusDungeonScore = 52
mythicPlusBestRunLevel = nil
(*temporary) = <table> {
 1 = "|cffc0c0c0"
 2 = "70 "
 3 = " "
 4 = "|cffffffff"
 5 = "Orc"
 6 = " "
 7 = "|cff00ff98"
 8 = "Monk"
 9 = "
|cffffd100"
 Index = 3
 count = 9
}
(*temporary) = <function> defined =[C]:-1
(*temporary) = "Mythic+ Rating: %s"
(*temporary) = "|cffffffff52|r"
(*temporary) = nil
(*temporary) = "ffffffff"
(*temporary) = "|cffffffff52|r"
(*temporary) = "attempt to concatenate a nil value"
UnitName = <function> defined =[C]:-1
cfg = <table> {
 iconCombat = true
 anchorWorldTipType = "parent"
 classification_elite = "+%s "
 iconFaction = true
 optionsBottom = 574.333069
 left = 492.500061
 optionsLeft = 822.167908
 anchorFrameTipType = "normal"
 classification_trivial = "~%s "
 classification_worldboss = "%s|r (Boss) "
 anchorFrameTipPoint = "BOTTOMLEFT"
 classification_rareelite = "+%s|r (Rare) "
 anchorFrameUnitPoint = "BOTTOMRIGHT"
 anchorWorldTipPoint = "BOTTOMRIGHT"
 colorNameByClass = false
 iconClass = true
 colRace = "|cffffffff"
 anchorWorldUnitType = "parent"
 targetYouText = "<<YOU>>"
 classification_normal = "%s "
 anchorOverrideCFPoint = "BOTTOMLEFT"
 showGuildRank = false
 colorGuildByReaction = true
 classification_rare = "%s|r (Rare) "
 top = 412.166656
 classification_minus = "-%s "
 enableAnchorOverrideCF = false
 anchorFrameUnitType = "normal"
 anchorWorldUnitPoint = "BOTTOMRIGHT"
}
UnitIsWildBattlePet = <function> defined =[C]:-1
UnitIsBattlePetCompanion = <function> defined =[C]:-1
lineInfo = <table> {
 1 = "|cffc0c0c0"
 2 = "70 "
 3 = " "
 4 = "|cffffffff"
 5 = "Orc"
 6 = " "
 7 = "|cff00ff98"
 8 = "Monk"
 9 = "
|cffffd100"
 Index = 3
 count = 9
}
tt = TipTac {
 BottomLeftCorner = Texture {
 }
 close = Button {
 }
 AddModifiedTip = <function> defined @TipTac/ttCore.lua:2153
 MirrorAnchorsSmart = <table> {
 }
 OnBackdropLoaded = <function> defined @SharedXML/Backdrop.lua:152
 GetNearestPixelSize = <function> defined @TipTac/ttCore.lua:620
 GetBackdropColor = <function> defined @SharedXML/Backdrop.lua:390
 RightEdge = Texture {
 }
 CVAR_UPDATE = <function> defined @TipTac/ttCore.lua:547
 ClearBackdrop = <function> defined @SharedXML/Backdrop.lua:282
 SetBackdropColorLocked = <function> defined @TipTac/ttCore.lua:796
 ADDON_LOADED = <function> defined @TipTac/ttCore.lua:2040
 GetDisplayedUnit = <function> defined @TipTac/ttCore.lua:364
 SetBackdropBorderColorLocked = <function> defined @TipTac/ttCore.lua:811
 ResetBackdropBorderColorLocked = <function> defined @TipTac/ttCore.lua:826
 tipsToModify = <table> {
 }
 SetupTextureCoordinates = <function> defined @SharedXML/Backdrop.lua:214
 SendElementEvent = <function> defined @TipTac/ttCore.lua:394
 HasBackdropInfo = <function> defined @SharedXML/Backdrop.lua:278
 GetEdgeSize = <function> defined @SharedXML/Backdrop.lua:188
 TopEdge = Texture {
 }
 App
```

**Comment by frozn** [comment]
Thanks for reporting this! 👍 

Fixed with release [v22.12.15](https://github.com/frozn/TipTac/releases/tag/v22.12.15-release).

The error showed up if there was no successful finished best mythic+ run available.

**Comment by Vilogity** [comment]
No Problems anymore :-)
You are so Fu***** fast ;-)
